gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/inspection/inspectResponses.ts

```diff
@@ -25,13 +39,118 @@ export default class OCR2InspectResponses extends SolanaCommand {
+  - Observer Count: ${transmission.observerCount}
+  - Observers: ${transmission.observers}
+  - Juels Per Lamport: ${transmission.juelsPerLamport}
+  - Reimbursement Gjuels: ${transmission.reimbursementGjuels}
+  `)
+        // Log responding oracle count
+        logger.info(`${transmission.observerCount}/${transmitters.length} oracles are responding
+        `)
+        // Log oracles that are not responsive
+        transmitters.forEach((transmitter) => {
+          // If the transmitter is not listed as an observer, log it
+          if (!transmission.observers.includes(transmitter)) {
+            logger.error(`Oracle ${transmitter} not responding
+            `)
+          }
+        })
+      }
+    }
+
+    // Define coder and event parser
+    const coder = new BorshCoder(idl)
+    const eventParser = new EventParser(programId, coder)
+    // Get block
+    let block = await this.provider.connection.getBlock(blockNumber)
+    if (!block) {
+      throw new Error('Block not found. Could not find latest block number in config')
+    }
+    // Iterate over all transactions in block
+    block.transactions.forEach((transaction) => {
+      // Get list of accounts keys associated with txn
+      let accountKeys = transaction.transaction.message.accountKeys.map((key) => `${key}`)
+      // Get list of instructions associated with txn
+      let instructions = transaction.transaction.message.instructions
+      // Check each instruction for program id and account
+      instructions.forEach((instruction) => {
+        // Instruction's program id == inputted program id
+        let hasProgramId = accountKeys[instruction.programIdIndex] == programId.toString()
+        // Transaction's account keys contains inputted account key
+        let hasAccount = accountKeys.includes(programAccount.toString())
+        // Parse event logs if above conditions are met
+        if (hasProgramId && hasAccount && transaction.meta && transaction.meta.logMessages) {
+          eventParser.parseLogs(transaction.meta.logMessages, logTransmissionEvent)
+        }
+      })
+    })
+  }
+
   execute = async () => {
     const ocr2 = getContract(CONTRACT_LIST.OCR_2, '')
     const program = this.loadProgram(ocr2.idl, ocr2.programId.toString())
 
     const state = new PublicKey(this.args[0])
     const input = this.makeInput(this.flags.input)
-    const data = await program.account.state.fetch(state)
+    const onChainState = await program.account.state.fetch(state)
+
+    logger.info(`Latest Config: 
+    - Latest Transmitter: ${onChainState.config.latestTransmitter}
+    - Latest Aggregator Round ID: ${onChainState.config.latestAggregatorRoundId}
+    - Latest Config Digest: ${onChainState.config.latestConfigDigest}
+    - Latest Config Block Number: ${onChainState.config.latestConfigBlockNumber}
+    `)
+
+    this.getLatestTransmissionEvent(
+      onChainState.config.latestConfigBlockNumber.toNumber(),
```

Comment:
- We want to inspect the latest transmission, for that we should search for the latest `NewTransmissions` event emitted. Now you are searching within the block where the latest config was set, this has 2 issues:
  - A new transmission was not necessarily included there
  - Most probably won't be the latest transmission in the contract
- `getLatestTransmissionEvent` should just get the transmission event. It should not log it or any other side effect.
- We can use anchor to parse the event, take a look at our SDK for parsing transmissions: https://github.com/smartcontractkit/chainlink-solana/blob/develop/ts/src/index.ts#L42